scripts/ci/in_container/update_quarantined_test_status.py

```diff
@@ -0,0 +1,226 @@
+    for line in body.splitlines(keepends=False):
+        if line.startswith("|-"):
+            parse = True
+            continue
+        if parse:
+            if not line.startswith("|"):
+                break
+            # noinspection PyBroadException
+            try:
+                status = parse_test_history(line)
+            except Exception:
+                continue
+            if status:
+                test_history_map[status.test_id] = status
+    return test_history_map
+
+
+def update_test_history(history: TestHistory, last_status: bool):
+    print(f"Adding status to test history: {history}, {last_status}")
+    return TestHistory(
+        test_id=history.test_id,
+        name=history.name,
+        url=history.url,
+        states=([last_status] + history.states)[0:num_runs]
+    )
+
+
+def create_test_history(result: TestResult) -> TestHistory:
+    print(f"Creating test history {result}")
+    return TestHistory(
+        test_id=result.test_id,
+        name=result.name,
+        url=get_url(result),
+        states=[result.result]
+    )
+
+
+def get_history_status(history: TestHistory):
+    if len(history.states) < num_runs:
+        if all(history.states):
+            return "So far, so good"
+        return "Flaky"
+    if all(history.states):
+        return "Stable"
+    if all(history.states[0:num_runs - 1]):
+        return "Just one more"
+    if all(history.states[0:int(num_runs / 2)]):
+        return "Almost there"
+    return "Flaky"
+
+
+def get_table(history_map: Dict[str, TestHistory]) -> str:
+    headers = ["Test", "Last run", f"Last {num_runs} runs", "Status"]
+    the_table: List[List[str]] = []
+    for ordered_key in sorted(history_map.keys()):
+        history = history_map[ordered_key]
+        the_table.append([
+            history.url,
+            "Succeeded" if history.states[0] else "Failed",
+            " ".join([reverse_status_map[state] for state in history.states]),
+            get_history_status(history)
+        ])
+    return tabulate(the_table, headers, tablefmt="github")
+
+
+if __name__ == '__main__':
+    if len(sys.argv) < 2:
+        print("Provide XML JUNIT FILE as first argument")
+        sys.exit(1)
+
+    with open(sys.argv[1], "r") as f:
+        text = f.read()
+    y = BeautifulSoup(text, "html.parser")
+    res = y.testsuites.testsuite.findAll("testcase")
+    for test in res:
+        print("Parsing: " + test['classname'] + "::" + test['name'])
+        if len(test.contents) > 0 and test.contents[0].name == 'skipped':
+            print(f"skipping {test['name']}")
+            continue
+        test_results.append(TestResult(
+            test_id=test['classname'] + "::" + test['name'],
+            file=test['file'],
+            line=test['line'],
+            name=test['name'],
+            classname=test['classname'],
+            result=len(test.contents) == 0
+        ))
+
+    token = os.environ.get("GITHUB_TOKEN")
+    print(f"Token: {token}")
```

Comment:
Should this be printed?

Reply:
Does not matter. It is masked anyway :)

Reply:
Just by seing ***** you know you get it right. girthub automatically mask github token when it is printed 😱

Reply:
See " https://github.com/apache/airflow/pull/10119/checks?check_run_id=938181654#step:6:1155

Reply:
> girthub automatically mask github token when it is printed 😱

I didn't know that. If that's the case I see no problem